src/net_processing.cpp

```diff
@@ -3786,7 +3786,7 @@ bool PeerManager::MaybeDiscourageAndDisconnect(CNode& pnode)
     if (pnode.addr.IsLocal()) {
         // We disconnect local peers for bad behavior but don't discourage (since that would discourage
         // all peers on the same local address)
-        LogPrintf("Warning: disconnecting but not discouraging local peer %d!\n", peer_id);
+        LogPrint(BCLog::NET, "Warning: disconnecting but not discouraging local peer %d!\n", peer_id);
```

Comment:
We can use `pnode.ConnectedThroughNetwork() == NET_ONION` to check for an onion peer.

Actually, if the tor daemon is running on another machine or running on the same machine but bitcoind<->tord communication is happening via some of the external (real) addresses, then incoming tor connections will not have `pnode->addr` equal to `127.0.0.1` and thus `pnode.addr.IsLocal()` will be `false`.

Reply:
>  thus pnode.addr.IsLocal() will be false.

That seems like something that can be improved unrelated to a logging-only change?

> We can use pnode.ConnectedThroughNetwork() == NET_ONION to check for an onion peer.

Onion peers are just an example of peers that may use the local proxy. I haven't tested, but it should be possible to use a proxy for ipv4 and ipv6 as well.

Reply:
> That seems like something that can be improved unrelated to a logging-only change?

Yes.

> Onion peers are just an example of peers that may use the local proxy.

You mean incoming peers to use a local proxy? I guess that would be possible, though maybe unusual.

Reply:
Added a check for `pnode.m_inbound_onion` for local onion proxies. I'll leave non-local onion proxies for a follow-up.